lib/_stream_writable.js

```diff
@@ -579,7 +578,7 @@ Writable.prototype.end = function(chunk, encoding, cb) {
   }
 
   // ignore unnecessary end() calls.
-  if (!state.ending && !state.finished)
+  if (!state.ending)
```

Comment:
This is actually a behavior change. Can you add a unit test?

Reply:
`state.finished = true` is only in `finishMaybe()`. If simplify the `finishMaybe()`, it should be:
```
function finishMaybe(stream, state) {
  if (state.ending) {
    state.finished = true;
  }
}
```
And, `state.ending` doesn't set to false.
So:
1. if `state.finished` is true, `state.ending` must be true:
        `!state.ending && !state.finished`  equals `false && false`. 
        `!state.ending` equals `false`. 
        `!state.ending && !state.finished` equal `!state.ending`.
2. if `state.finished` is false:
        `!state.ending && !state.finished` equals `!state.ending && true`. 
        `!state.ending && !state.finished` equals `!state.ending`.